### ISSUE TITLE
Port the code generators to run as maven plugins and make them public

### ISSUE THREAD

**Issue by david-at-aws** [amazon-archives/aws-sdk-java-resources#6]
All of the code in the service-specific modules is generated from [JSON files](https://github.com/awslabs/aws-sdk-java-resources/blob/master/aws-resources-ec2/src/main/resources/com/amazonaws/resources/ec2/model.json) that describe the resources for a service and how they relate together. Right now the code generators depend on some internal Amazon infrastructure, but we'd love to get them converted to run as Maven plugins so we can push them out to GitHub instead of just the code they generate. Comment here if that's something you'd like to see us prioritize!

**Comment by futurely** [comment]
It seems that you didn't use generic converters such as [jsonschema2pojo](https://github.com/joelittlejohn/jsonschema2pojo). I'd like to know how did you implement the generator and how to validate that the json schema is consistent with the services API specification. Even if you don't convert the generator into Maven plugins, it's very helpful to share the existing code.

**Comment by david-at-aws** [comment]
Correct, it's not a generic POJO converter, since the generated resource objects aren't normal POJOs. I got about halfway through getting the code generators into a shareable form a while back before I got distracted by re:Invent - I'll see if I can get that finished up in the next couple weeks. Thanks for the reminder!

**Comment by futurely** [comment]
Among the many RESTful API specifications such as RAML, API-blueprint and WADL etc., [Swagger](http://swagger.io/) is one of the most popular one that provides [comprehensive tool sets](https://github.com/swagger-api/swagger-spec) including maven plugins and is one of the API specifications supported by the [Microsoft Azure API management service](http://azure.microsoft.com/en-in/services/api-management/) (former APIphany). Swagger representes API specification as JSON as this project does.

The feature that is most relevant to this issue is the [code generator](https://github.com/swagger-api/swagger-codegen). It can generate [client, server and many more](https://github.com/swagger-api/swagger-codegen/tree/master/samples) in [most of the main stream languages](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources).

**Comment by futurely** [comment]
APIphany must have been sold to Microsoft at a very good price considering [the previous rounds of funding](https://angel.co/apiphany). Unfortunately, AWS doesn't have API management service yet.
